desc/optimize/tr_subproblems.py

```diff
@@ -370,6 +370,107 @@ def loop_body(state):
     return cond(jnp.linalg.norm(p_newton) <= trust_radius, truefun, falsefun, None)
 
 
+@jit
+def trust_region_step_exact_qr(
+    f, J, trust_radius, initial_alpha=None, rtol=0.01, max_iter=10
+):
+    """Solve a trust-region problem using a semi-exact method.
+
+    Solves problems of the form
+        min_p ||J*p + f||^2,  ||p|| < trust_radius
```

Comment:
```suggestion
    .. math::
        \min_p ||J*p + f||^2,  ||p|| < trust_radius
```
in this way it will be formatted using math-latex style